account_budget_oca/models/account_budget.py

```diff
@@ -0,0 +1,234 @@
+    _name = "crossovered.budget"
+    _description = "Budget"
+    _inherit = ['mail.thread']
+
+    name = fields.Char(
+        string='Budget Name', required=True,
+        states={'done': [('readonly', True)]})
+    creating_user_id = fields.Many2one(
+        comodel_name='res.users', string='Responsible',
+        default=lambda self: self.env.user)
+    date_from = fields.Date(
+        string='Start Date', required=True,
+        states={'done': [('readonly', True)]})
+    date_to = fields.Date(
+        string='End Date', required=True,
+        states={'done': [('readonly', True)]})
+    state = fields.Selection(
+        selection=[('draft', 'Draft'),
+                   ('cancel', 'Cancelled'),
+                   ('confirm', 'Confirmed'),
+                   ('validate', 'Validated'),
+                   ('done', 'Done')],
+        string='Status', default='draft', index=True, required=True,
+        readonly=True, copy=False, track_visibility='always')
+    crossovered_budget_line_ids = fields.One2many(
+        comodel_name='crossovered.budget.lines',
+        inverse_name='crossovered_budget_id', string='Budget Lines',
+        states={'done': [('readonly', True)]}, copy=True)
+    company_id = fields.Many2one(
+        comodel_name='res.company', string='Company', required=True,
+        default=lambda self: self.env['res.company']._company_default_get(
+            'account.budget.post'))
+
+    @api.multi
+    def action_budget_confirm(self):
+        self.write({'state': 'confirm'})
+
+    @api.multi
+    def action_budget_draft(self):
+        self.write({'state': 'draft'})
+
+    @api.multi
+    def action_budget_validate(self):
+        self.write({'state': 'validate'})
+
+    @api.multi
+    def action_budget_cancel(self):
+        self.write({'state': 'cancel'})
+
+    @api.multi
+    def action_budget_done(self):
+        self.write({'state': 'done'})
+
+
+class CrossoveredBudgetLines(models.Model):
+    _name = "crossovered.budget.lines"
+    _description = "Budget Line"
+
+    crossovered_budget_id = fields.Many2one(
+        comodel_name='crossovered.budget', string='Budget', ondelete='cascade',
+        index=True, required=True)
+    analytic_account_id = fields.Many2one(
+        comodel_name='account.analytic.account', string='Analytic Account')
+    general_budget_id = fields.Many2one(
+        comodel_name='account.budget.post', string='Budgetary Position',
+        required=True)
+    date_from = fields.Date(string='Start Date', required=True)
+    date_to = fields.Date(string='End Date', required=True)
+    paid_date = fields.Date(string='Paid Date')
+    planned_amount = fields.Float(
+        string='Planned Amount', required=True, digits=0)
+    practical_amount = fields.Float(
+        compute='_compute_practical_amount', string='Practical Amount',
+        digits=0)
+    theoretical_amount = fields.Float(
+        compute='_compute_theoretical_amount', string='Theoretical Amount',
```

Comment:
`string='Theoretical Amount',` could be removed, since Odoo will set it automatically.
See https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#fields

Reply:
OK